statistics/handle.go

```diff
@@ -126,27 +126,28 @@ func (h *Handle) Update(is infoschema.InfoSchema) error {
 
 	tables := make([]*Table, 0, len(rows))
 	deletedTableIDs := make([]int64, 0, len(rows))
+	pid2tid := buildPartitionID2TableID(is)
```

Comment:
I think we need to cache the map in `Handle`, only rebuild when the infochema has changed.